### ISSUE TITLE
Flatpak issue when using Vorta to mount a local backup on Mint 19.1 

### ISSUE THREAD

**Issue by goebbe** [linuxmint/mintinstall#254]
After installing the Flatpak build of Vorta there is the following issue in Linux Mint 19.1 : 
Local backups cannot be mounted via Vorta (using fuse).

Background:
- Vorta https://github.com/borgbase/vorta  is a relatively new backup client with a GUI - it is basically a graphical interface for borgbackup.  
- For a couple of weeks Vorta provides a Flatpak-build via flathub. 
Testing the build I realized that Vorta was not able to mount my local backup on LM 19.1.

Since I never had problems using the integrated flatpak-support in LM, I opened the following issue for Vorta: https://github.com/borgbase/vorta/issues/328

However, the developer could not reproduce the issue, so I started testing different distributions in a virtual machine. I always used the same flatpak-build (from flathub).
**Here is a short summary:** 
Linux Mint 19.1 (Cinnamon - based on Ubuntu 18.04 LTS) <- issue appears
Linux Mint Debian Edition 3 (Cinnamon - based on Debian "Stretch") in virtual machine <- no issue
Linux Mint Xfce  19.1 (based on Ubuntu 18.04. LTS) in virtual machine <- issue appears
Ubuntu 18.04.2 LTS (using https://flatpak.org/setup/Ubuntu/) in virtual machine <- no issue
The Developer tested other distribution without issue. 

In short the issue has only be reproduced on Linux Mint (based on Ubuntu 18.04). 

To reproduce, do the following: 
1. Install Vorta via Flathub
2. In Repository: Create a new local repository for backups (with default settings) 
3. In Sources: Select a folder to be backed up
4. Click on "Start Backup" to create an archive in the repository
5. In Archives: Select an archive (i.e. backup) and click "Mount" and select a mount point.

Result: 
- In Vorta Archive-Tab, in the line of the repository, at the column "Mount Point": My selected mountpoint appears
- In the Nemo - File Manager:
  the chosen mountpoint(folder) disappears
  the mountpoint appears in the side pane (as mounted)
  Now - if I select the mount point in the side pane I get the following error message:
   "The folder contents could not be displayed. Sorry, could not display all the content of "mountpoint": **Error opening directory '/home/myuser/mountpoint': Transport endpoint is not connected"**
   The folder is not displayed.
- When I type `ls -All` now, **in a console**, I get the following output:
`d????????? ? ? ? ? ? mountpoint`
- Unmounting the repository (via Vorta) brings everything back to normal.

Expected behaviour: The repository should be mounted at the mountpoint. i.e. you can use the file browser to explore the backup.

Mounting via vorta works well on LMDE 3 and Ubuntu 18.04.2.!

Additional remarks: 
borgbackup is using FUSE to mount the encrypted repositories/archives. 
I suspect differences in the configuration/ Integration of flatpak is the root of the problem.

For more details I would like to point you to the original issue.

Sorry for the long report. :-(
Please let me know, if I can provide additional information.

**Comment by goebbe** [comment]
Since Linux Mint 19.1 (based on Ubuntu 18.04) with flatpak 1.0.8 did not work, in the next step I tried the following: 
- Install Linux Mint 19.1 in a virtal machine
- remove --purge flatpak (1.0.8).
- install flatpak 1.4.2 from the ppa of alexlarson: ppa:alexlarsson/flatpak
- update and upgrade all pakages and reboot
Result: the issue is still present.
The issue is not due to a specific bug in flatpak 1.0.8. - so it seems to be something specific about Linux Mint

**Comment by RavetcoFX** [comment]
can you try reproducing this on mint 18.3?

**Comment by goebbe** [comment]
I can reproduce on Mint 18.3:
Linux Mint 18.3 (Cinnamon - based on Ubuntu 16.04 LTS) <- issue appears

Just for the record:
Debian Buster (new stable, Cinnamon) <- no issue

**Comment by RavetcoFX** [comment]
that's interesting. Does a non flatpak version work?

**Comment by clefebvre** [comment]
Are xdg-desktop-portal and xdg-desktop-portal-gtk installed?

**Comment by goebbe** [comment]
@RavetcoFX  I had problems building the Vorta, that is the reason I tested the flatpak version. I will, again, try to build Vorta as a non flapak version and report, here.
- What works as a non-flapak version is the use of borgbackup on the console and the use of the  commands `borg mount `and `borg umount` (that use FUSE in the background in order to mount the repository).
@clefebvre yes, xdg-desktop-portal and xdg-desktop-portal-gtk are both installed. 
It is always the default install + updates (if necessary installation of flatpak) + installation of Vorta

**Comment by goebbe** [comment]
@RavetcoFX now I build/ installed a **non flatpak** version of Vorta: 
The non flatpak build works **without issues**.

For the record: Steps to install Vorta on Linux Mint 19.1 (non flatpak): 
```
install pip3:
sudo apt install python3-pip
pip3 install setuptools
pip3 install wheel
pip3 install vorta
```

**Comment by goebbe** [comment]
I just tested Beta of Mint 19.2:
Linux Mint Beta 19.2 (Cinnamon - based on Ubuntu 16.04 LTS) <- issue appears

**Comment by goebbe** [comment]
Now I tried to get additional "debugging" information.

```
#open a shell insides the flatpak environment: 
flatpak run --command=sh  com.borgbase.Vorta 

[📦 com.borgbase.Vorta ~]$ borg --version
borg 1.1.10

[📦 com.borgbase.Vorta ~]$ borg mount --debug  /home/goebbe/myrepository /home/goebbe/mountpoint
using builtin fallback logging configuration
35 self tests completed in 0.12 seconds
Verified integrity of /home/goebbe/myrepository/index.5
Enter passphrase for key /home/goebbe/myrepository: 
TAM-verified manifest
security: read previous location '/home/goebbe/myrepository'
security: read manifest timestamp '2019-07-17T08:35:58.321972'
security: determined newest manifest timestamp as 2019-07-17T08:35:58.321972
security: repository checks ok, allowing access
mount data cache capacity: 4 chunks
Mounting filesystem
Initializing llfuse
Calling fuse_mount
Calling fuse_lowlevel_new
Calling fuse_session_add_chan

[📦 com.borgbase.Vorta ~]$ ls mo*
ls: cannot access 'mountpoint': Transport endpoint is not connected
```

Hmm - not very helpful. :-/       I am really lost, here.  
Fuse does not seem to mount / hand over correctly. But why is this issue only present in LM 18.3-19.2 and not in LMDE 3 or Debian Buster (cinnamon)  or Ubuntu 18.04?

**Comment by RavetcoFX** [comment]
I have a feeling its something related to the kernel version being used. Can you post what kernel you're on with uname -a and try updating the kernel to a later version with mintupdate

**Comment by goebbe** [comment]
@RavetcoFX   I can confirm, that this issue is related to the kernel. 
Now, in Linux Mint 19.2 beta I tested the following kernels: 

 uname -a
Linux goebbe-Standard-PC-i440FX-PIIX-1996 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux        <- issue appears

 uname -a
Linux goebbe-Standard-PC-i440FX-PIIX-1996 5.0.0-20-generic #21~18.04.1-Ubuntu SMP Thu Jun 27 04:04:37 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux       <- no issue

 uname -a
Linux goebbe-Standard-PC-i440FX-PIIX-1996 4.18.0-25-generic #26~18.04.1-Ubuntu SMP Thu Jun 27 07:28:31 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux    <- no issue

To summarize: The issue appears with kernel  4.15.0-54-generic  but not with kernel 4.18 and 5.0

A workaround for this issue would be to install a newer kernel.

**Comment by goebbe** [comment]
I just double checked, why Ubuntu 18.04 (+ flatpak ppa) did work without issue, while Mint 19.1 did not: 
I downloaded the latest desktop version of Ubuntu 18.04 - that was version 18.04.2.  
As compared with original Ubuntu 18.04.0, this version comes with the updated linux kernel 4.18.0
This explains why my Ubuntu 18.04.2 worked but Mint 19.1 (with kernel 4.15.0) did not.

**Comment by goebbe** [comment]
I am closing this issue, since apparently the issue has been caused by the specific combination of flatpak/ fuse/ linux kernel. 
There are several ways to workaround: 
- Vorta can be installed directly - as outlined in the comments above, or
- upgrading the Kernel from 4.15 to a newer one, or
- distribution upgrade to Linux mint 21 or newer
